src/Fl_x.cxx

```diff
@@ -1276,7 +1253,15 @@ int fl_handle(const XEvent& thisevent)
     Fl_X11_Screen_Driver *d = (Fl_X11_Screen_Driver*)Fl::screen_driver();
     d->init_workarea();
 #if USE_XFT
-    after_display_rescale(&(d->current_xft_dpi));
+    float old_scales[Fl::screen_count()];
+    for (int i = 0; i < Fl::screen_count(); i++) old_scales[i] = d->scale(i);
+    d->use_startup_scale_factor();
+    for (int i = 0; i < Fl::screen_count(); i++) {
+      float new_scale = d->scale(i);
+      if (old_scales[i] - new_scale < 0.01) continue;
```

Comment:
Hmm, shouldn't this compare the absolute value of the difference? The removed code seemed to use `fabs()` in a similar case...
Sorry, if this is wrong, I'm just having a quick look at the code.

Reply:
good catch. fixed